system_test/aest_sync_SUITE.erl

```diff
@@ -0,0 +1,190 @@
+-module(aest_sync_SUITE).
+
+%=== EXPORTS ===================================================================
+
+% Common Test exports
+-export([all/0]).
+-export([init_per_testcase/2]).
+-export([end_per_testcase/2]).
+
+% Test cases
+-export([new_node_joins_network/1,
+         docker_keeps_data/1,
+         crash_and_continue_sync/1]).
+
+-import(aest_nodes, [
+    setup_nodes/2,
+    start_node/2,
+    stop_node/2, stop_node/3,
+    kill_node/2,
+    http_get/5,
+    request/4,
+    wait_for_height/4,
+    assert_synchronized/2
+]).
+
+%=== INCLUDES ==================================================================
+
+-include_lib("eunit/include/eunit.hrl").
+
+%=== MACROS ====================================================================
+
+% -define(MINING_TIMEOUT, 4*10000).
+-define(MINING_TIMEOUT, 1000).
+
+-define(NODE1, #{
+    name    => node1,
+    peers   => [node2],
+    backend => aest_docker,
+    source  => {pull, "aetrnty/epoch:v0.8.0"}
+}).
+
+-define(NODE2, #{
+    name    => node2,
+    peers   => [node1],
+    backend => aest_docker,
+    source  => {pull, "aetrnty/epoch:v0.8.0"}
+}).
+
+-define(NODE3, #{
+    name    => node3,
+    peers   => [node1],
+    backend => aest_docker,
+    source  => {pull, "aetrnty/epoch:local"}
+}).
+
+%=== COMMON TEST FUNCTIONS =====================================================
+
+all() -> [
+    new_node_joins_network
+    % , docker_keeps_data
+    , crash_and_continue_sync
+].
+
+init_per_testcase(_TC, Config) ->
+    %% Some parameters depend on the speed and capacity of the docker containers:
+    aest_nodes:ct_setup([ {blocks_per_second, 3},
+                          {node_startup_time, 5000}, %% Time it takes to get the node to respond to http
+                          {node_stop_time, 20000}    %% Time it takes to get the node to stop cleanly
+                          | Config]).
+
+end_per_testcase(_TC, Config) ->
+    aest_nodes:ct_cleanup(Config).
+
+%=== TEST CASES ================================================================
+
+%% A few tests that verify that our assumptions are right for docker timings and
+%% API.
+
+new_node_joins_network(Cfg) ->
+    Length = 10,
+    BlocksPerSecond = proplists:get_value(blocks_per_second, Cfg),
+    NodeStartupTime = proplists:get_value(node_startup_time, Cfg), 
+
+    setup_nodes([?NODE1, ?NODE2, ?NODE3], Cfg),
+    start_node(node1, Cfg),
+    start_node(node2, Cfg),
+    wait_for_height(Length, [node1, node2], Length * ?MINING_TIMEOUT, Cfg),
+    Top1 = request(node1, [v2, 'top'], #{}, Cfg),
+    ct:log("Node 1 top: ~p~n", [Top1]),
+    Height = request(node2, [v2, 'block-by-height'], #{height => Length}, Cfg),
+    ct:log("Node 2 at height ~p: ~p~n", [Length, Height]),
+    start_node(node3, Cfg),
+    wait_for_height(0, [node3], NodeStartupTime, Cfg),
+    ct:log("Node 3 ready to go"),
+
+    %% get them with N blocks a second
+    timer:sleep((Length div BlocksPerSecond) * 1000),
```

Comment:
Is this test actually meant to fail if sync is slower than a threshold, or is this meant to be large timeout?
If performance-related, maybe such tests shall be tagged as "performance" or similar and skipped unless explicitly enabled.

Reply:
It is not performance related, I should probably change the timing to something like the `docker_keeps_data` test where I measure the average time for mining a block. The test is about having a node with the last code join a cluster of older version nodes and check it synchronize and not start a new chain.